SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/RefreshActorDormancyTest.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "RefreshActorDormancyTest.h"
+
+#include "Engine/NetDriver.h"
+#include "Engine/NetworkObjectList.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Net/UnrealNetwork.h"
+#include "RefreshActorDormancyTestActor.h"
+
+ARefreshActorDormancyTest::ARefreshActorDormancyTest()
+{
+	Author = TEXT("Ajanth Sutharsan");
+}
+
+void ARefreshActorDormancyTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, DormantToAwakeActor);
+	DOREPLIFETIME(ThisClass, AwakeToDormantActor);
+}
+
+void ARefreshActorDormancyTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Step 1 - Spawn dormancy actor which is non dormant and dormant initially and then become dormant and non dormant respectively
+	AddStep(TEXT("ServerSpawnDormantAndNonDormatActors"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		DormantToAwakeActor = SpawnActor<ARefreshActorDormancyTestActor>(FActorSpawnParameters(), true);
+		AwakeToDormantActor = SpawnActor<ARefreshActorDormancyTestActor>(FActorSpawnParameters(), true);
+		DormantToAwakeActor->SetInitiallyDormant(true);
+		AwakeToDormantActor->SetInitiallyDormant(false);
+		FinishStep();
+	});
+
+	// Step 2 - Client check NetDormancy is DORM_Awake for DormantToAwakeActor and DORM_DormantAll for AwakeToDormantActor
+	AddStep(("ClientAssertDormancyTestStates"), FWorkerDefinition::AllClients, nullptr, nullptr,
+			[this](float DeltaTime) {
+				RequireEqual_Int(DormantToAwakeActor->NetDormancy, DORM_Awake, TEXT("DormantToAwakeActor is Awake"));
```

Comment:
Should add a nullptr check for both of these incase the client ticks before they have received the replicated actors